autoload/ale/list.vim

```diff
@@ -151,6 +156,31 @@ function! s:SetListsImpl(timer_id, buffer, loclist) abort
     endif
 endfunction
 
+" Try to restore the window view after closing any of the lists to avoid making
+" the it moving around, especially useful when on insert mode
+function! s:RestoreViewIfNeeded(buffer) abort
+    let l:buffer = getbufvar(str2nr(a:buffer), '', {})
```

Comment:
```suggestion
    let l:saved_view = getbufvar(a:buffer, 'ale_winview', {})
```

You can get the view variable right away, with a default, and you can check `empty(l:saved_view)` No need for `str2nr` here too. The buffer variable should already be a number.

Reply:
Done!